homework/index.js

```diff
@@ -29,14 +29,86 @@
     });
     return elem;
   }
+  function addOptions(select, repoNames) {
+    repoNames.forEach(repoName => createAndAppend('option', select, { text: repoName.name }));
+  }
 
   function main(url) {
     fetchJSON(url, (err, data) => {
       const root = document.getElementById('root');
       if (err) {
         createAndAppend('div', root, { text: err.message, class: 'alert-error' });
       } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        const repoNames = data.sort((a, b) =>
+          a.name.toLowerCase().localeCompare(b.name.toLowerCase()),
+        );
+
+        const header = createAndAppend('header', root, { id: 'header' });
```

Comment:
The `id` attributes that you set in several places in your code are unneeded. Initially, you had them in there because you needed them for use with `getElementById()`. Now that you have changed your code as per my feedback to use JavaScript variables for accessing the elements created with `createAndAppend` you no longer need them (and therefore you should remove them).

You only need to add an `id` if:

1. You still need, for some reason, to use `getElementById` to get a reference to the element.
2. You need the `id` for CSS styling:

   ```css
   #header {
     ...

If none of these cases apply (and I don't think they do in your case) you should remove the redundant `id` attributes. Leaving them in will leave people looking at your code puzzled why they are there.